client_cli/src/main.rs

```diff
@@ -1,86 +1,97 @@
-        let content = fs::read_to_string(file).wrap_err(err_msg)?;
-        let metadata: UnlimitedMetadata = json5::from_str(&content).wrap_err(deser_err_msg)?;
-        Ok(Self(metadata))
-    }
-}
+impl MetadataArgs {
+    fn load(self) -> Result<UnlimitedMetadata> {
+        let value: Option<UnlimitedMetadata> = self
+            .metadata
+            .map(|path| {
+                let content = fs::read_to_string(&path).wrap_err_with(|| {
+                    eyre!("Failed to read the metadata file `{}`", path.display())
+                })?;
+                let metadata: UnlimitedMetadata =
+                    json5::from_str(&content).wrap_err_with(|| {
+                        eyre!(
+                            "Failed to deserialize metadata from file `{}`",
+                            path.display()
+                        )
+                    })?;
+                Ok::<_, color_eyre::Report>(metadata)
+            })
+            .transpose()?;
 
-/// Client configuration wrapper. Allows getting itself from arguments from cli (from user supplied file).
-#[derive(Debug, Clone)]
-struct Configuration(pub ClientConfiguration);
-
-impl FromStr for Configuration {
-    type Err = Error;
-    fn from_str(file: &str) -> Result<Self> {
-        let deser_err_msg = format!("Failed to decode config file {} ", &file);
-        let err_msg = format!("Failed to open config file {}", &file);
-        let content = fs::read_to_string(file).wrap_err(err_msg)?;
-        let cfg = json5::from_str(&content).wrap_err(deser_err_msg)?;
-        Ok(Self(cfg))
+        Ok(value.unwrap_or_default())
     }
 }
 
 /// Iroha CLI Client provides an ability to interact with Iroha Peers Web API without direct network usage.
-#[derive(StructOpt, Debug)]
-#[structopt(name = "iroha_client_cli", version = concat!("version=", env!("CARGO_PKG_VERSION"), " git_commit_sha=", env!("VERGEN_GIT_SHA")), author)]
+#[derive(clap::Parser, Debug)]
+#[command(name = "iroha_client_cli", version = concat!("version=", env!("CARGO_PKG_VERSION"), " git_commit_sha=", env!("VERGEN_GIT_SHA")), author)]
 struct Args {
-    /// Sets a config file path
-    #[structopt(short, long)]
-    config: Option<Configuration>,
+    /// Path to the configuration file, defaults to `config.json`/`config.json5`
+    ///
+    /// Supported extensions are `.json` and `.json5`. By default, Iroha looks up for a
+    /// `config` file in the Current Working Directory with both supported extensions.
+    /// If the default config file is not found, Iroha Client will rely on default values and
+    /// environment variables. However, if the config path is set explicitly with this argument
+    /// and the file is not found, Iroha Client will exit with an error.
+    #[arg(
+        short,
+        long,
+        value_name("PATH"),
+        value_hint(clap::ValueHint::FilePath),
+        value_parser(Path::user_provided_str)
+    )]
+    config: Option<Path>,
```

Comment:
@outoftardis @yamkovoy please take a look.

(Essentially the same text as in `cli/src/main.rs`)